sphinx/config.py

```diff
@@ -201,6 +194,13 @@ def __init__(self, dirname, filename, overrides, tags):
                     config[k] = copyright_year_re.sub(r'\g<1>%s' % format_date('%Y'),
                                                       config[k])
 
+    @classmethod
+    def read(cls, confdir, filename, overrides=None, tags=None):
+        # type: (unicode, unicode, Dict, Tags) -> Config
+        """Create a Config object from configuration file."""
+        namespace = eval_config_file(confdir, filename, tags)
+        return Config(namespace, overrides or {})
```

Comment:
You should probably use `cls` here.

    return cls(namespace, overrides or {})

This also means people can subclass this if necessary.

Reply:
Absolutely

Reply:
fixed. (will be pushed later)